### PR TITLE
Allow custom objects to use leaderboard properties

### PR DESCRIPTION
**Pull request by D8H**

### CODE REVIEW

File: newIDE/app/src/CompactPropertiesEditor/index.js [hunk 1]
```diff
@@ -998,6 +1009,9 @@ const CompactPropertiesEditor = ({
         return contentView;
       } else if (field.valueType === 'resource') {
         return renderResourceField(field);
+      } else if (field.valueType === 'leaderboardId') {
+        // TODO Make a compact version of LeaderboardIdPropertyField
+        return null;
```
Comment (D8H) [inline newIDE/app/src/CompactPropertiesEditor/index.js:1014]: I think it needs to be designed first.
The property won't be shown in the side panel. I guess, it's not really an issue.

File: newIDE/app/src/Leaderboard/UseLeaderboardReplacer.js [hunk 2]
```diff
@@ -260,10 +260,10 @@ export const replaceLeaderboardsInProject = async ({
         project.getProjectUuid(),
         {
           sourceGameId: sourceGameId,
-          sourceLeaderboardId: leaderboardId.replace(/"/g, ''),
+          sourceLeaderboardId: leaderboardId,
         }
       );
-      replacedLeaderboardsMap[leaderboardId] = `"${duplicatedLeaderboard.id}"`;
+      replacedLeaderboardsMap[leaderboardId] = duplicatedLeaderboard.id;
```
Comment (D8H) [inline newIDE/app/src/Leaderboard/UseLeaderboardReplacer.js:266]: Leaderboard identifiers are no longer expressions but only the identifier.

File: newIDE/app/src/CompactPropertiesEditor/PropertiesMapToCompactSchema.js [hunk 3]
```diff
@@ -164,6 +164,23 @@ const createField = (
       getDescription,
       hasImpactOnAllOtherFields: property.hasImpactOnOtherProperties(),
     };
+  } else if (valueType === 'leaderboardid') {
```
Comment (4ian) [inline newIDE/app/src/CompactPropertiesEditor/PropertiesMapToCompactSchema.js:167]: the type has a typo

Comment (D8H) [reply newIDE/app/src/CompactPropertiesEditor/PropertiesMapToCompactSchema.js:167]: `valueType` is lower case.
```JS
  const valueType = property.getType().toLowerCase();
```

**Comment by 4ian** [comment]
Notes for myself before we merge this:
- investigate making a compact field for it (select at a minimum + button to administrate, with a menu to choose between "Create/edit leaderboards" and "Enter manually")
- that's it?

**Review by AlexandreSi — APPROVED** [review]
LGTM

**Pull request merged**